src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -479,14 +521,19 @@ private void BeginCondenseTransition()
             if (this.IsExpanded == false)
                 throw new InvalidOperationException();
 
-            RefreshCondensedDisplay();
+            //To prevent another transition from being started
+            currentState = State.InTransition;
```

Comment:
I think `SetCurrentStateAndNotify(State.InTransition);` was made for this purpose, and I remember not to set the `this.currentState` directly. The same for another one below. Does that cause any problem?